internal/nginx/maxmind.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nginx
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+)
+
+// MaxmindLicenseKey maxmind license key to download databases
+var MaxmindLicenseKey = ""
+
+const (
+	geoIPPath = "/etc/nginx/geoip"
+
+	geoLiteCityDB = "GeoLite2-City"
+	geoLiteASNDB  = "GeoLite2-ASN"
```

Comment:
Commercial GeoIP2 databases are more accurate than GeoLite2 ones. We use `GeoIP2-ISP` instead of `GeoLite2-ASN`. So with this design I had to create a placeholder `GeoLite2-ASN.mmdb` file to trick the controller into believing this file exist so that it does not force disable `use-geoip2` setting.

I think instead of this the controller should get the comma separated list of DB file names as a command line arrgument and then check for existence of them.

Reply:
I think we should be more dramatic than that. Right now those files are present in the [docker image](https://github.com/kubernetes/ingress-nginx/tree/master/images/nginx/rootfs/etc/nginx/geoip) and are thirteen months old. The old geoip module databases are not maintained anymore.
Before any change, I think we should consider drop support for the old geoip module.